File: task/src/main/java/com/goexp/galgame/data/task/handler/DownloadGameHandler.java
package com.goexp.galgame.data.task.handler;

import com.goexp.galgame.data.piplline.core.Message;
import com.goexp.galgame.data.piplline.core.MessageQueueProxy;
import com.goexp.galgame.data.piplline.handler.DefaultMessageHandler;
import com.goexp.galgame.data.task.client.GetChu;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class DownloadGameHandler extends DefaultMessageHandler<Integer> {

    final private Logger logger = LoggerFactory.getLogger(DownloadGameHandler.class);


    @Override
    public void process(final Message<Integer> message, MessageQueueProxy<Message> msgQueue) {

        var gid = message.entity;
        logger.debug("Download {}", gid);


        try {
            GetChu.GameService.download(gid);
        } catch (Exception e) {
            logger.error("Re-down:{}", gid);
            msgQueue.offer(new Message<>(MesType.NEED_DOWN_GAME, gid));
            e.printStackTrace();
        }

        msgQueue.offer(new Message<>(MesType.Game, gid));

    }

}

File: task/src/main/scala/com/goexp/galgame/data/task/handler/DownloadGameHandler.scala
package com.goexp.galgame.data.task.handler

import com.goexp.galgame.data.piplline.core.{Message, MessageHandler}
import com.goexp.galgame.data.task.client.GetChu.GameService.Download
import com.goexp.galgame.data.task.handler.game.Bytes2Html
import org.slf4j.LoggerFactory

/**
  * Net IO
  */
class DownloadGameHandler extends MessageHandler {
  final private val logger = LoggerFactory.getLogger(classOf[DownloadGameHandler])

  override def process(message: Message) = {
    message.entity match {

      case gid: Int =>
        logger.debug("Download {}", gid)
        try {
          val zipBytes = Download.getBytes(gid)
          logger.debug("Download OK:{}", gid)

          send(Message(classOf[Bytes2Html].hashCode(), (gid, zipBytes)))
        }
        catch {
          case e: Exception =>
            logger.error("Re-down:{} IOException:{}", gid, e.getMessage)
            send(Message(classOf[DownloadGameHandler].hashCode(), gid))

        }
    }

    //    val (gid: Int) = message.entity

  }
}